pandas/core/indexes/base.py

```diff
@@ -1311,6 +1312,52 @@ def _get_names(self):
         return FrozenList((self.name, ))
 
     def _set_names(self, values, level=None):
+        """
+        Set new names on index.
+
+        Parameters
+        ----------
+        values : str or sequence
+            name(s) to set
+        level : int, level name, or sequence of int/level names (default None)
+            If the index is a MultiIndex (hierarchical), level(s) to set (None
+            for all levels).  Otherwise level must be None
+
+        Returns
+        -------
+        index : Index
+
+        See Also
+        --------
+        Index.set_names : Set new names on index. Defaults to returning
+        new index.
+        Index.rename : Set new names on index. Defaults to returning new index.
+
+        Notes
+        -----
+        Both `set_names` and `rename` call this function to set name.
+
+        Examples
+        --------
+        on an index with no names:
```

Comment:
you don't need the fulll doc string here (e.g. examples and such, leave Parameters and such), only on ``.set_names``